integration/spark/app/src/test/java/io/openlineage/spark/agent/SparkStreamingTest.java

```diff
@@ -0,0 +1,344 @@
+
+            assertEquals(expectedOutputSchema, outputSchemaFields);
+          });
+    }
+
+    private List<SchemaRecord> mapToSchemaRecord(OpenLineage.SchemaDatasetFacet schema) {
+      return schema.getFields().stream()
+          .map(field -> new SchemaRecord(field.getName(), field.getType()))
+          .collect(Collectors.toList());
+    }
+
+    private void populateTopic(String bootstrapServers, String topic) {
+      Properties props = new Properties();
+      props.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, bootstrapServers);
+      props.put(
+          ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, StringSerializer.class.getCanonicalName());
+      props.put(
+          ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, StringSerializer.class.getCanonicalName());
+      props.put(ProducerConfig.LINGER_MS_CONFIG, "100");
+
+      ObjectMapper om = new ObjectMapper().findAndRegisterModules();
+      try (KafkaProducer<String, String> producer = new KafkaProducer<>(props)) {
+        List<Future<RecordMetadata>> futures =
+            IntStream.range(0, 100)
+                .mapToObj(
+                    ignored ->
+                        new InputMessage(
+                            UUID.randomUUID().toString(), Instant.now().getEpochSecond()))
+                .map(message -> serialize(om, message))
+                .map(json -> new ProducerRecord<String, String>(topic, json))
+                .map(
+                    x ->
+                        producer.send(
+                            x,
+                            (ignored, e) -> {
+                              if (e != null) {
+                                log.error("Failed to publish a message", e);
+                              }
+                            }))
+                .collect(Collectors.toList());
+
+        for (Future<RecordMetadata> future : futures) {
+          future.get();
+        }
+      } catch (ExecutionException | InterruptedException e) {
+        throw new RuntimeException(e);
+      }
+    }
+
+    @SneakyThrows
+    private String serialize(ObjectMapper mapper, InputMessage message) {
+      return mapper.writeValueAsString(message);
+    }
+
+    private void createTopics(String bootstrapServers, Collection<String> topics) {
+      try (AdminClient adminClient =
+          AdminClient.create(
+              ImmutableMap.of(AdminClientConfig.BOOTSTRAP_SERVERS_CONFIG, bootstrapServers))) {
+        List<NewTopic> newTopics =
+            topics.stream()
+                .distinct()
+                .map(topicName -> new NewTopic(topicName, 1, (short) 1))
+                .collect(Collectors.toList());
+
+        CreateTopicsResult result = adminClient.createTopics(newTopics);
+        result.all().get();
+      } catch (ExecutionException | InterruptedException e) {
+        throw new RuntimeException(e);
+      }
+    }
+  }
+}
+
+@Tag("integration-test")
```

Comment:
Is this tag necessary? 
I really like this testing approach 👍

Reply:
I think it was left by accident :) I ve played with some stuff and forgot to remove it